java/java.editor/src/org/netbeans/modules/editor/java/JavaCompletionCollector.java

```diff
@@ -689,32 +689,63 @@ public Completion createStaticMemberItem(CompilationInfo info, DeclaredType type
             });
             String label = type.asElement().getSimpleName() + "." + memberElem.getSimpleName();
             String sortText = memberElem.getSimpleName().toString();
+            String memberTypeName;
+            StringBuilder labelDetail = new StringBuilder();
+            StringBuilder insertText = new StringBuilder(label);
+            boolean asTemplate = false;
             if (memberElem.getKind().isField()) {
+                memberTypeName = Utilities.getTypeName(info, memberType, false).toString();
                 sortText += String.format("#%s", Utilities.getTypeName(info, type, false)); //NOI18N
             } else {
+                memberTypeName = Utilities.getTypeName(info, ((ExecutableType) memberType).getReturnType(), false).toString();
```

Comment:
Nitpick: maybe guard that memberType.getKind() is a method/constructor/executable.